we should have a config-based way to skip flaky tests

zdevito opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#30978 we should have a config-based way to skip flaky tests**

This particular approach queries our issue tracker for test titles that
match the following format:

```
DISABLED test_async_grad_guard_with_grad (jit.test_async.TestAsync)
```

And then skips the python test for them. There is 1 second timeout so
if the internet flakes we still run the test suite, without disabling any
tests.

This is intended as a quick fix, similar to ninja unland, to get to a green
master. Long term test disables should go into the code.

Test Plan: Imported from OSS

Pulled By: zdevito

Differential Revision: [D18890532](https://our.internmc.facebook.com/intern/diff/D18890532)

zdevito committed:
we should have a config-based way to skip flaky tests

This particular approach queries our issue tracker for test titles that
match the following format:

```
DISABLED test_async_grad_guard_with_grad (jit.test_async.TestAsync)
```

And then skips the python test for them. There is 1 second timeout so
if the internet flakes we still run the test suite, without disabling any
tests.

This is intended as a quick fix, similar to ninja unland, to get to a green
master. Long term test disables should go into the code.

Test Plan: Imported from OSS

Pulled By: zdevito

kostmo commented:
## CircleCI build failures summary
As of commit cb910cf:
* **1/1** failures introduced in this PR



## Detailed failure analysis (WIP)

Here are the [reasons each build failed](https://dr.pytorch.org/commit-details.html?sha1=cb910cf5763500cc1fd758033b11d15828de9b35):

### 1 failure recognized by patterns:
#### [![See CircleCI build](https://avatars0.githubusercontent.com/ml/7?s=12)](https://circleci.com/gh/pytorch/pytorch/3857182) pytorch_linux_xenial_py3_6_gcc5_4_build (1/1)
**Step:** Build ([details](https://dr.pytorch.org/build-details.html?build_id=70605958))
```
Automatic merge failed; fix conflicts and then commit the result.
```


---
This comment was automatically generated by [Dr. CI](https://github.com/kostmo/circleci-failure-tracker/tree/master/docs/from-pull-request-comment).
Follow [this link to opt-out](https://dr.pytorch.org/admin/comments-opt-out.html) of these comments for your Pull Requests.


Please report bugs/suggestions on the [GitHub issue tracker](https://github.com/kostmo/circleci-failure-tracker/issues).

zdevito committed:
Update on "we should have a config-based way to skip flaky tests"

we should have a config-based way to skip flaky tests

This particular approach queries our issue tracker for test titles that
match the following format:

```
DISABLED test_async_grad_guard_with_grad (jit.test_async.TestAsync)
```

And then skips the python test for them. There is 1 second timeout so
if the internet flakes we still run the test suite, without disabling any
tests.

This is intended as a quick fix, similar to ninja unland, to get to a green
master. Long term test disables should go into the code.

Test Plan: Imported from OSS

Pulled By: zdevito

gh-metadata: pytorch pytorch 30978 gh/zdevito/138/head
